apps/studio/components/interfaces/TableGridEditor/GridHeaderActions.tsx

```diff
@@ -142,6 +144,8 @@ const GridHeaderActions = ({ table }: GridHeaderActionsProps) => {
           </Button>
         )}
 
+        {roleImpersonationEnabledFlag && <RoleImpersonationPopover serviceRoleLabel="postgres" />}
```

Comment:
once i select to impersonate a user - on a macbook viewport, it's overflowing already
![image](https://github.com/supabase/supabase/assets/19742402/715d9a0c-23f2-4d30-b3f6-7142a2f84d1d)

Terry mentioned something related [here](https://github.com/supabase/supabase/pull/18945#discussion_r1406677954) i think - not sure if this is something we should tackle in this PR cc @MildTomato. The quick win here might be to make the data/definition smaller - instead of toggle tabs, we use a dropdown or smth

Reply:
https://www.notion.so/supabase/Table-Editor-Header-ce7f90895596411ab7533d8e7952eb37?pvs=4#603240cb8d334a9fb63efc23a8d0abfe

these are the mockups to consolidate the UI

Reply:
Let's fix in a follow up PR 🙌🏻